src/code/z_collision_check.c

```diff
@@ -2969,33 +2997,36 @@ static s8 sBssDummy12;
 static s8 sBssDummy13;
 static s8 sBssDummy14;
 
-void func_800628A4(s32 arg0, ColliderJntSph* collider) {
+// Updates the world spheres for all of the collider's JntSph elements attached to the specified limb
+void Collider_UpdateSpheres(s32 limb, ColliderJntSph* collider) {
     static Vec3f D_8015E648;
     static Vec3f D_8015CF00; // bss ordering changes here
     s32 i;
 
     for (i = 0; i < collider->count; i++) {
-        if (arg0 == collider->list[i].dim.joint) {
-            D_8015E648.x = collider->list[i].dim.modelSphere.center.x;
-            D_8015E648.y = collider->list[i].dim.modelSphere.center.y;
-            D_8015E648.z = collider->list[i].dim.modelSphere.center.z;
+        if (limb == collider->elements[i].dim.limb) {
+            D_8015E648.x = collider->elements[i].dim.modelSphere.center.x;
+            D_8015E648.y = collider->elements[i].dim.modelSphere.center.y;
+            D_8015E648.z = collider->elements[i].dim.modelSphere.center.z;
             Matrix_MultVec3f(&D_8015E648, &D_8015CF00);
-            collider->list[i].dim.worldSphere.center.x = (s32)D_8015CF00.x;
-            collider->list[i].dim.worldSphere.center.y = (s32)D_8015CF00.y;
-            collider->list[i].dim.worldSphere.center.z = (s32)D_8015CF00.z;
-            collider->list[i].dim.worldSphere.radius =
-                (s32)((f32)collider->list[i].dim.modelSphere.radius * collider->list[i].dim.scale);
+            collider->elements[i].dim.worldSphere.center.x = D_8015CF00.x;
+            collider->elements[i].dim.worldSphere.center.y = D_8015CF00.y;
+            collider->elements[i].dim.worldSphere.center.z = D_8015CF00.z;
+            collider->elements[i].dim.worldSphere.radius =
+                collider->elements[i].dim.modelSphere.radius * collider->elements[i].dim.scale;
         }
     }
 }
 
-void func_80062A28(GlobalContext* globalCtx, Vec3f* v) {
+// Spawns red blood droplets.
+// No actor has a collision type that spawns red blood.
+void CollisionCheck_SpawnRedBlood(GlobalContext* globalCtx, Vec3f* v) {
     static EffectSparkInit D_8015CF10;
-    s32 sp24;
```

Comment:
should be simple enough to name this blood or something

Reply:
oh i meant the D_ symbol specifically, the temp is the effect index so bloodIndex or something makes sense for that

Reply:
I'm leaving all the bss symbols as D_ for now because of the current bss ordering issues in the file. Since they're all static, it'll be easy to fix them later.

Reply:
ah thats fair, np